### PR TITLE
#5 Listing

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-id-mobile/issues/5 |
| Decisions | - Add listing component based of components-vue listing and adapted to work with new ripe-id account listing. |
| Animated GIF | image |

### CODE REVIEW

**Review by joao-conde — APPROVED** [review]

**Comment by BeeMargarida** [comment]
@3rdvision Are you adapting to use getItems logic?

File: react/components/organisms/listing/listing.stories.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                placeholder: "Dummy Filter",
+                width: 150,
+                options: [
+                    { label: "Value 1", value: 1 },
+                    { label: "Value 2", value: 2 },
+                    { label: "Value 3", value: 3 }
+                ]
+            }
+        ];
+        const filtersValue = { role: null, company: null, nationality: null };
```
Comment (gcandal) [inline react/components/organisms/listing/listing.stories.js:69]: Shouldn't be mandatory

File: react/components/organisms/listing/listing.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            onRefresh: PropTypes.func,
+            onEndReached: PropTypes.func,
+            style: ViewPropTypes.style
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            items: [],
+            renderItem: () => {},
```
Comment (gcandal) [inline react/components/organisms/listing/listing.js:28]: It either is required or has a default prop

File: react/components/organisms/listing/listing.js [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+    ViewPropTypes
+} from "react-native";
+import PropTypes from "prop-types";
+
+import { Search, Select } from "../../molecules";
+
+export class Listing extends Component {
+    static get propTypes() {
+        return {
+            items: PropTypes.array,
```
Comment (gcandal) [inline react/components/organisms/listing/listing.js:18]: Dead code?

File: react/components/organisms/listing/listing.stories.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                    avatarURL={item.avatar_url}
+                    title={item.meta.name}
+                    key={index}
+                />
+            );
+        };
+
+        return (
+            <View style={{ flex: 1, paddingVertical: 15, backgroundColor: "#f6f7f9" }}>
+                <Listing items={items} renderItem={renderItem} filters={filters} />
```
Comment (gcandal) [inline react/components/organisms/listing/listing.stories.js:82]: Use getItems instead (you can just return items).

File: react/components/organisms/listing/listing.js (hunk 3 above)
Comment (3rdvision) [reply react/components/organisms/listing/listing.js:18]: This is not dead code.
You can either use getItems to fetch data or items directly and use the callbacks to control listing refresh from the outside.
We will lose flexibility if we remove this.

Comment (gcandal) [reply react/components/organisms/listing/listing.js:18]: This listing must follow the same approach as the listing from ripe-components-vue. If you want to use a fixed set of items you can just `getItems={() => items}`

**Review by gcandal — APPROVED** [review]

File: react/components/organisms/listing/listing.js [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+                refreshing: false
+            });
+        });
+    }
+
+    scrollToTop = () => {
+        this.flatListRef.scrollToOffset({ animated: true, offset: 0 });
+    };
+
+    _getItems = async (options = {}) => {
```
Comment (joamag) [inline react/components/organisms/listing/listing.js:88]: let's move these two methods down to the private methods section

**Review by joamag — CHANGES_REQUESTED** [review]

**Pull request merged**